src/main/java/com/hubspot/jinjava/el/ext/AstRangeBracket.java

```diff
@@ -72,6 +72,17 @@ public Object eval(Bindings bindings, ELContext context) {
     PyList result = new PyList(new ArrayList<>());
     int index = 0;
 
+    // Handle negative indeces.
+    if ((startNum < 0) || (endNum < 0)) {
+       int size = Iterables.size(baseItr);
```

Comment:
If it's cheaper enough to use array length here I can special case that..

Reply:
`Iterables.size()` has some smart logic to determine the fastest way to calculate the size depending on the underlying iterable implementation IIRC.

Reply:
Sorry, ignore me.  Arrays turn into lists which are cheap for calculating size...